### DISCUSSION TITLE
Source files in src/peripherals orphaned from CMake

### DISCUSSION THREAD

**Discussion by dmeehan1968** [Lora-net/LoRaMac-node#1363] [Q&A]
There are number of .c and .h files in src/peripherals (e.g. gpio-ioe.c/h, mag3110.c/h, ...) the headers for which are included into some of the board specific implementatons, but they are not included in CMakeLists.txt files which means the C files would not be built and linked.  This seems to make them orphaned, unless there are definitions in the headers that are referenced.

Any reason why they are still part of the repo?

**Comment by mluis1** [comment]
The NAMote72 target requires some of those files. This is the reason why they are still present.
All files under `src/peripherals` are included for the build process.

Please refer to below lines and mainly to `"*.c"`:
https://github.com/Lora-net/LoRaMac-node/blob/55f83f90ddbc4bc6aabd34a3262d3870548b3577/src/peripherals/CMakeLists.txt#L26
https://github.com/Lora-net/LoRaMac-node/blob/55f83f90ddbc4bc6aabd34a3262d3870548b3577/src/peripherals/CMakeLists.txt#L30
https://github.com/Lora-net/LoRaMac-node/blob/55f83f90ddbc4bc6aabd34a3262d3870548b3577/src/peripherals/CMakeLists.txt#L35

**Reply by dmeehan1968** [reply]
The lines you point too do include those files regardless of which secure element is selected (I missed the glob pattern when checking if they were orphaned).

But that means they are being included in projects that aren't using them, relying on dead code elimination at link to remove the redundancy.  Might there be a case for moving the inclusion into the board specific CMakeLists.txt even if the source modules remain in the peripheral directory?

I'm working on the static library build, and I'm trying to avoid copying over legacy stuff (if its redundant)

**Reply by dmeehan1968** [reply]
i'm going to query this again, for example:

- gpio-ioe.c/h: Included in several board implementations but the function usage is conditional on BOARD_IOE_EXT which is never referenced in a CMakeLists.txt (granted the developer could opt to define globally, and I'm not sure what functionality this represents)
- mag3110.c/h: Included in NAMote72 board.c but none of the defines or functions are referenced
- mma8451.c/h: Included in NAMote72 board.c, perhaps should be move local to this board implementation?
- mpl3113.c/h: Included in NAMote72 board.c, fuota-test-01/NAMote72/main.c, periodic-uplink-lpp/NAMote72/main.c, , perhaps should be move local to this board implementation?
- pam7q.c/h: Never included, no symbols referenced
- sx1509.c/h: Included by gpio-ioe.h, referenced in gpio-ioe.c, so also dependent on BOARD_IOE_EXT
- sx9500.c/h: Included in NAMote72 board.c, perhaps should be move local to this board implementation?

It seems that most are for NAMote72 support and should be local to the board implementation for that.  The gpio-ioe inclusion seems dependent on a conditional which may no longer be relevant as its clearly not used by any of the examples.  Use case needs to be defined.
